buildSrc/src/main/kotlin/ServerConfiguration.kt

```diff
@@ -0,0 +1,118 @@
+ *
+ * Please contact JetBrains, Na Hrebenech II 1718/10, Prague, 14000, Czech Republic
+ * if you need additional information or have any questions.
+ */
+
+import org.gradle.api.Project
+import org.gradle.api.Task
+import org.gradle.api.artifacts.Dependency
+import org.gradle.api.artifacts.dsl.DependencyHandler
+import org.gradle.api.file.DuplicatesStrategy
+import org.gradle.api.plugins.JavaApplication
+import org.gradle.api.tasks.JavaExec
+import org.gradle.api.tasks.bundling.Jar
+import org.gradle.api.tasks.compile.JavaCompile
+import org.gradle.kotlin.dsl.*
+import java.util.*
+
+public fun Project.applyCommonServerConfiguration(javaApplication: JavaApplication) {
+
+  val launcherClassName = "org.jetbrains.projector.server.ProjectorLauncher"
+
+  javaApplication.apply {
+    mainClass.set(launcherClassName)
+  }
+
+  publishToSpace()
+
+  version = project(":projector-server-common").version
+
+  dependencies {
+    api(project(":projector-server-common"))
+  }
+
+  tasks.named<Jar>("jar") {
+    manifest {
+      attributes(
+        "Main-Class" to launcherClassName,
+      )
+    }
+    duplicatesStrategy = DuplicatesStrategy.WARN
+  }
+
+  // Server running tasks
+  val localProperties = Properties()
+  if (rootProject.file("local.properties").canRead()) {
+    localProperties.load(rootProject.file("local.properties").inputStream())
+  }
+
+  // Relay arguments
+  val relayURL = localProperties["ORG_JETBRAINS_PROJECTOR_SERVER_RELAY_URL"]
+  val serverId = localProperties["ORG_JETBRAINS_PROJECTOR_SERVER_RELAY_SERVER_ID"]
+
+  var relayArgs = emptyList<String>()
+
+  if (relayURL != null && serverId != null) {
+    relayArgs = listOf("-DORG_JETBRAINS_PROJECTOR_SERVER_RELAY_URL=$relayURL", "-DORG_JETBRAINS_PROJECTOR_SERVER_RELAY_SERVER_ID=$serverId")
+  }
+
+  fun JavaExec.setupHeadlessServer() {
+    jvmArgs(relayArgs)
+  }
+
+  val targetJvmVersion = tasks.withType(JavaCompile::class.java).firstOrNull()?.sourceCompatibility
+  checkNotNull(targetJvmVersion) { "Module target JDK version not set" }
+
+  fun Task.checkJavaVersion() {
+    doFirst {
+      check(Runtime.version().feature() >= targetJvmVersion.toInt()) {
+        "Task '$name' in module '${this@applyCommonServerConfiguration.name}' requires JDK version $targetJvmVersion or newer. Current version - ${Runtime.version()}"
+      }
+    }
+  }
+
+  createRunServerTask(
+    name = "runServer",
+    isAgent = false,
+    localProperties,
+  ) {
+    checkJavaVersion()
+    jvmArgs(
+      "--add-opens=java.desktop/java.awt=ALL-UNNAMED",
+      "--add-opens=java.desktop/sun.font=ALL-UNNAMED",
+      "--add-opens=java.base/java.lang.reflect=ALL-UNNAMED",
+    )
+    setupHeadlessServer()
+  }
+
+  createRunIdeaTask(
+    name = "runIdeaServer",
+    isAgent = false,
+    localProperties,
+  ) {
+    checkJavaVersion()
+    setupHeadlessServer()
+  }
+}
+
+private fun DependencyHandler.api(dependencyNotation: Any): Dependency? =
+  add("api", dependencyNotation)
```

Comment:
I remember we had some similar problems when extracting code to buildSrc. Please at least add todo here.

@Avvessalom, can I ask you to research this problem of unavailability of some symbols in buildSrc?